flox-bash/lib/commands.sh

```diff
@@ -108,6 +108,43 @@ function parseNixArgs() {
 	done
 }
 
+# This function parses out common flake options used by the flox
+# builders as used by several of the environment builder and
+# development subcommands.
+declare -a _floxFlakeArgs
+function parseFloxFlakeArgs() {
+	_floxFlakeArgs=()
+	_cmdArgs=()
+	while test $# -gt 0; do
+		case "$1" in
+		# Options taking two args.
+		--arg|--argstr|--override-flake|--override-input)
+			_floxFlakeArgs+=("$1"); shift
+			_floxFlakeArgs+=("$1"); shift
+			_floxFlakeArgs+=("$1"); shift
+			;;
+		# Options taking one arg.
+		--priority|--eval-store|--include|-I|--inputs-from|--update-input|--expr|--file|-f)
+			_floxFlakeArgs+=("$1"); shift
+			_floxFlakeArgs+=("$1"); shift
+			;;
+		# Options taking zero args.
+		--debugger|--impure|--commit-lock-file|--no-registries|--no-update-lock-file|--no-write-lock-file|--recreate-lock-file|--derivation)
+			_floxFlakeArgs+=("$1"); shift
+			;;
+		# Consume remaining args
+		--command|-c|--)
+			_cmdArgs+=("$@")
+			break
+			;;
```

Comment:
we got this twice now; also in https://github.com/flox/flox/pull/110/files#diff-f2b7203d41a3b7ded798568a5d9c980434118874c941694821bfa15599a2ffe2R99

Reply:
This is correct; each time we call either of `parse{NixArgs,FloxFlakeArgs}()` we need to return by way of `${_cmdArgs[@]}` any args following a `--command` or `--` arg. Think of this as a passthru.